vars/DocMultiplicationPipeline.groovy

```diff
@@ -0,0 +1,413 @@
+def call(Map config = [:]) {
+    def params = config.params ?: [:]
+    def contexts = config.contexts ?: [:]
+    def workerAgent = config.workerAgent ?: 'Jenkins-Default-Agent-X64-C5xlarge-Single-Host'
+    
+    node(workerAgent) {
+        try {
+            echo "Starting Document Multiplication Pipeline"
+            echo "Target: Create large snapshot in S3 bucket: ${params.snapshotBucketPrefix}${params.stage}"
+            
+            env.AWS_DEFAULT_REGION = "${params.region}"
+            env.STAGE = "${params.stage}"
+            env.DEBUG_MODE = "${params.debugMode}"
+            
+            stage('1. Checkout') {
```

Comment:
These first few stages should be the same for our default and this pipeline. Can we do something like this to retain the common logic: 
```
// vars/buildStage.groovy
def call() {
    stage("Build") {
        steps {
            sh "echo Building..."
        }
    }
}
Then in your pipelines:


@Library('my-shared-lib') _
pipeline {
    agent any
    stages {
        stage("Common") {
            steps {
                buildStage()
            }
        }
    }
}
```